### PR TITLE
Forms: Add multiline input

### PR DESCRIPTION
**Pull request by dennisreimann**
Up to now we don't have a way to accept multiline input values via the forms. The input type text strips the newlines and replaces them with whitespace, so there wasn't even a way to solve #4941 correctly.

This add the textarea form provider and modifies the PosData partial, so that it properly displays multiline values.

Closes #4941.

### CODE REVIEW

File: BTCPayServer/Views/Shared/PosData.cshtml [hunk 1]
```diff
@@ -26,13 +28,13 @@
                     }
                     else
                     {
-                        @value?.ToString()
+                        @Safe.Raw(Regex.Replace(str.Trim(), @"(\r\n)|\n|\r", "<br/>"))
```
Comment (dennisreimann) [inline BTCPayServer/Views/Shared/PosData.cshtml:31]: @NicolasDorier OK like this?

Comment (learntheropes) [reply BTCPayServer/Views/Shared/PosData.cshtml:31]: Guys, sorry for asking, but I don't understand anything of this programming language.

I just want to make sure that the replacement happens only in the rendered field and does not manipulate the data actually stored in the metadata property, which are correct at the moment.

If we recall the invoice, we should find the same text we have added when we created it.

Comment (Kukks) [reply BTCPayServer/Views/Shared/PosData.cshtml:31]: This seems excessively complicated, I think we should do as suggested with `white-space: pre-wrap;` styling

**Review by Kukks — CHANGES_REQUESTED** [review]

File: BTCPayServer/Views/Shared/PosData.cshtml (hunk 1 above)
Comment (dennisreimann) [reply BTCPayServer/Views/Shared/PosData.cshtml]: Done in [5501b91](https://github.com/btcpayserver/btcpayserver/pull/4942/commits/5501b916897bfd63489260979d01cae558ba003e).

**Review by dstrukt — APPROVED** [review]
LGTM!

At least visually / functionally.

File: BTCPayServer/Views/Shared/Forms/TextareaElement.cshtml [hunk 2]
```diff
@@ -0,0 +1,28 @@
+@model BTCPayServer.Abstractions.Form.Field
+@{
+	var isInvalid = ViewContext.ModelState[Model.Name]?.ValidationState is Microsoft.AspNetCore.Mvc.ModelBinding.ModelValidationState.Invalid;
+	var errors = isInvalid ? ViewContext.ModelState[Model.Name].Errors : null;
+}
+<div class="form-group">
+    <label class="form-label" for="@Model.Name"@(Model.Required ? " data-required" : "")>
+        @Model.Label
+    </label>
+    <textarea id="@Model.Name" type="@Model.Type" class="form-control @(errors is null ? "" : "is-invalid")"
```
Comment (dennisreimann) [inline BTCPayServer/Views/Shared/Forms/TextareaElement.cshtml:10]:
```suggestion
    <textarea id="@Model.Name" class="form-control @(errors is null ? "" : "is-invalid")"
```

**Comment by dennisreimann** [comment]
@Kukks Your feedback has been addressed. Can we merge it so that I can use the textarea in #4695?

**Review by Kukks — APPROVED** [review]

**Pull request merged**